src/main/java/org/ballerinalang/containers/docker/BallerinaDockerClient.java

```diff
@@ -80,6 +80,23 @@ String createMainImage(String packageName, String dockerEnv, List<Path> bPackage
             throws BallerinaDockerClientException, IOException, InterruptedException;
 
     /**
+     * Create a Ballerina Main Docker image from a Ballerina BAL / BALX File.
+     *
+     * @param packageName  The name used to identify the package.
+     * @param dockerEnv    The docker host URL.
+     * @param bPackagePath The paths to package to be packed to the image.
+     * @param imageName    The docker image name to use.
+     * @param imageVersion The docker image version to use.
+     * @return The {@link String} name of the docker image created. Null if the image building process failed.
+     * @throws BallerinaDockerClientException If the input parameters are invalid.
+     * @throws IOException                    If specified files cannot be accessed.
+     * @throws InterruptedException           If the docker image build process was interrupted.
+     */
+    String createMainImage(String packageName, String dockerEnv, Path bPackagePath,
```

Comment:
Not sure if this is the correct way to tackle this. 

1. Why cannot we keep a single method accepting a `List<Path>` for package paths and handle the single package path case inside?
2. Shouldn't we deprecate the main and service image creation methods since they are not to be used anymore?

Reply:
+1, I had the same in my mind. we need to deprecate the main and service image creation methods.

Reply:
+1 to a single method. And I think we could directly remove the methods, as this is not being exposed as an API yet.